src/theory/quantifiers/candidate_rewrite_filter.cpp

```diff
@@ -135,6 +135,10 @@ bool MatchTrie::getMatches(Node n, NotifyMatch* ntm)
               recurse = false;
             }
           }
+          else if (!var.getType().isSubtypeOf(cn.getType()))
```

Comment:
I imagine you would not want to merge this `if` condition with the one above it? As is it can be a little annoying to see when `recurse` is actually gonna be false. Also, you could avoid the lookup in the map if you tested the type first, no?

Reply:
Computing types is actually (much) more expensive than a map lookup.  The code is organized such that this is minimized.

Reply:
I was afraid that was the case. :)